scripts/data_collection/git_repo.py

```diff
@@ -0,0 +1,32 @@
+import os
+import subprocess
+
+
+class GitRepository:
+    def __init__(self, repository, project_dir):
+        self.username, self.project_name = repository.split('/')
+        self.project_dir = project_dir
+
+    def clone(self):
+        p = subprocess.Popen(
+            ["git", "clone", f"https://github.com/{self.username}/{self.project_name}.git",
+             os.path.basename(self.project_dir), "--depth", "1"], cwd=os.path.dirname(self.project_dir))
+        return_code = p.wait()
+        return return_code == 0
+
+    def pull_changes_without_history(self) -> bool:
```

Comment:
Maybe should we add an argument for the depth parameter and add an ability for users to choose the depth?